src/middlewared/middlewared/plugins/network.py

```diff
@@ -1916,9 +1916,11 @@ def ip_in_use(self, choices):
     @private
     def get_nic_names(self) -> set:
         """Get network interface names excluding internal interfaces"""
+        res, ignore = set(), set(self.middleware.call_sync('interface.internal_interfaces'))
         with scandir('/sys/class/net/') as nics:
-            res = set(nic.name for nic in nics)
-        ignore = set(self.middleware.call_sync('interface.internal_interfaces'))
+            for nic in filter(lambda x: x.is_symlink() and x.name not in ignore, nics):
+                res.add(nic.name)
+
         return res - ignore
```

Comment:
AFAICS, no longer need to subtract ignore (because of `and x.name not in ignore`)

Reply:
derp